### PR TITLE
[Patterns]: Prepare GB bundled patterns for core back port

### PR DESCRIPTION
**Pull request by ntsekouras**
## Description

This PR updates the GB pattern registration, as they are going to be included in `core`.

### CODE REVIEW

File: lib/block-patterns.php [hunk 1]
```diff
@@ -180,8 +156,15 @@ function register_gutenberg_patterns() {
...
 								<!-- wp:social-link {"url":"#","service":"mail"} /--></ul>
 								<!-- /wp:social-links -->',
-		)
+		),
 	);
+
+	foreach ( $patterns as $name => $pattern ) {
+		$pattern_name = 'core/' . $name;
+		if ( ! WP_Block_Patterns_Registry::get_instance()->is_registered( $pattern_name ) ) {
+			register_block_pattern( $pattern_name, $pattern );
```
Comment (gziolo) [inline lib/block-patterns.php:165]: @youknowriad should we unregister patterns from core if they exist? We do it for blocks.

**Comment by gziolo** [comment]
Nice one. It's better to guard it first before it causes issues with CI once it's in core 👍🏻

File: lib/block-patterns.php (hunk 1 above)
Comment (youknowriad) [reply lib/block-patterns.php:165]: Yes, I think it's better to override them.

**Comment by ntsekouras** [comment]
Back port to core PR: https://github.com/WordPress/wordpress-develop/pull/1274

File: lib/block-patterns.php (hunk 1 above)
Comment (ntsekouras) [reply lib/block-patterns.php:165]: Done here: https://github.com/WordPress/gutenberg/pull/32086/commits/a38ff60f13d204e9b8d1330f9ce01aaa0af03fc2

**Review by youknowriad — APPROVED** [review]

**Review by ryelle — APPROVED** [review]
👍🏻  This fixes the issue with duplicate block patterns

**Comment by youknowriad** [comment]
when folks do a patch release, please don't forget this PR then :)

**Comment by ntsekouras** [comment]
I can do a patch release for this PR now :)

**Pull request merged**